site/components/DateSelector.tsx

```diff
@@ -15,37 +16,62 @@ interface DateSelectorProps {
     inputId: string;
     inputName: string;
     disablePast: boolean;
+    pageState: PageState;
+    updatePageState: Dispatch<SetStateAction<PageState>>;
+    updaterFunction: (
+        currentState: PageState,
+        setPageState: Dispatch<SetStateAction<PageState>>,
+        inputName: keyof PageInputs,
+        input: string | Date | null,
+        error?: ErrorInfo,
+    ) => void;
 }
 
 const inputBox = (
-    errors: ErrorInfo[],
     inputRef: React.Ref<HTMLInputElement> | undefined,
     inputProps: InputBaseComponentProps | undefined,
     InputProps: Partial<FilledInputProps> | Partial<OutlinedInputProps> | undefined,
     disabled: boolean,
     inputId: string,
     inputName: string,
+    pageState: PageState,
+    updatePageState: Dispatch<SetStateAction<PageState>>,
+    updaterFunction: (
+        currentState: PageState,
+        setPageState: Dispatch<SetStateAction<PageState>>,
+        inputName: keyof PageInputs,
+        input: string | Date | null,
+        error?: ErrorInfo,
+    ) => void,
 ) => (
-    <FormElementWrapper errors={errors} errorId={inputId} errorClass="govuk-date-input--error">
-        <div className="govuk-date-input flex items-center [&_.MuiSvgIcon-root]:fill-govBlue">
-            <div className="govuk-date-input__item govuk-!-margin-right-0">
-                <div className="govuk-form-group">
-                    <input
-                        className={`govuk-input govuk-date-input__input govuk-input--width-6 ${
-                            errors.length > 0 ? "govuk-input--error" : ""
-                        } `}
-                        id={inputId}
-                        name={inputName}
-                        type="text"
-                        ref={inputRef}
-                        {...inputProps}
-                        disabled={disabled}
-                    />
-                </div>
-            </div>
-            {InputProps?.endAdornment}
+    <div className="govuk-date-input flex items-center [&_.MuiSvgIcon-root]:fill-govBlue">
+        <div className="govuk-date-input__item govuk-!-margin-right-0">
+            <FormElementWrapper errors={pageState.errors} errorId={inputId} errorClass="govuk-input--error">
+                <input
+                    className="govuk-input govuk-date-input__input govuk-input--width-6"
+                    id={inputId}
+                    name={inputName}
+                    type="text"
+                    ref={inputRef}
+                    {...inputProps}
+                    disabled={disabled}
+                    placeholder={disabled ? "N/A" : "DD/MM/YYYY"}
+                    onBlur={(e) => {
+                        const input = e.target.value;
+                        if (!input) {
+                            updaterFunction(pageState, updatePageState, inputId as keyof PageInputs, input, {
```

Comment:
If inputId needs to be a key of PageInputs can we not just enforce that in the props rather than using `as` here?

Reply:
Yep, changed.